applications/MetisApplication/custom_processes/metis_divide_heterogeneous_input_in_memory_process.h

```diff
@@ -24,9 +24,7 @@
 // Project includes
 #include "includes/model_part_io.h"
 #include "custom_processes/metis_divide_input_to_partitions_process.h"
-
-// This one needs mpi
-#include "mpi.h"
```

Comment:
I would include the data communicator include and leave the mpi one as well, otherwise you are already implying that `DataCommunicator` will always be `MpiDataCommunicator`.

Or either enforce the usage of the `MpiDataCommunicator`

Reply:
I am working on removing both those includes. Basically once the DataComm has an async interface we can completely drop the explicit MPI dependency which is what I am after :)

Reply:
(BTW MPIDataComm includes mpi.h)